### PR TITLE
Adding capability for plotting loss curves

### PR DESCRIPTION
**Pull request by namrathaurs**

### CODE REVIEW

File: src/deepdiagnostics/plots/lossPlot.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        raise NotImplementedError
+
+    def plot_name(self):
+        return "loss_curves.png"
+   
+    def plot(
+       self,
+       training_history: Dict[str, List[float]],
+       epochs: int = None,
+       best_val_loss: float = None
```
Comment (voetberg) [inline src/deepdiagnostics/plots/lossPlot.py:20]: Some possible things to set as options: 
* Train and val loss line style, color 
* Option to display the val loss or not

File: src/deepdiagnostics/plots/lossPlot.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    def _data_setup(self):
+        raise NotImplementedError
+
+    def plot_name(self):
+        return "loss_curves.png"
+   
+    def plot(
+       self,
+       training_history: Dict[str, List[float]],
```
Comment (voetberg) [inline src/deepdiagnostics/plots/lossPlot.py:18]: Because the keys here are set, you can set the type to be `dict[Literal["train_loss", "val_loss"]: list[float]]`

File: src/deepdiagnostics/plots/lossPlot.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        plt.plot(epochs_trained, validation_loss_data, label='Validation Loss')
+        plt.axhline(y=best_val_loss, color='m', linestyle='--', label="Best Val. Loss")
+
+        plt.xlabel('Epochs')
+        plt.ylabel('Loss')
+        plt.title('Training and Validation Loss Over Epochs')
+
+        plt.legend()
+        plt.grid()
+        plt.show()
```
Comment (voetberg) [inline src/deepdiagnostics/plots/lossPlot.py:52]: Two things: 
* Please use the super `_finish` to either save or show the plot
* Please return the figure, axes object so the user can edit them outside of the plot method if they want
